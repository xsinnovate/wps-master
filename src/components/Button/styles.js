import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    height: 40,
    borderColor: '#333',
    borderRadius: 20,
    borderWidth: 1,
    paddingHorizontal: 15,
  },
  text: {
    color: '#333',
    fontSize: 16,
    fontWeight: '600',
    lineHeight: 40,
    textAlign: 'center',
  },
});
