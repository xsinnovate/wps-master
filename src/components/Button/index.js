import PropTypes from 'prop-types';
import React from 'react';
import { TouchableHighlight, Text, View } from 'react-native';
import styles from './styles';

export default function Button({ children, onPress }) {
  return (
    <TouchableHighlight
      activeOpacity={0.5}
      onPress={onPress}
      underlayColor="#fff"
    >
      <View style={styles.container}>
        <Text style={styles.text}>
          {children}
        </Text>
      </View>
    </TouchableHighlight>
  );
}

Button.propTypes = {
  children: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};
