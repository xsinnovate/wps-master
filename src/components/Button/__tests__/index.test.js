import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableHighlight } from 'react-native';
import Button from '..';

const noop = () => { };

describe('<Button />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button onPress={noop}>Label</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls onPress prop when pressed', () => {
    const spy = jest.fn();
    const instance = renderer.create(<Button onPress={spy}>Label</Button>).root;

    instance.findByType(TouchableHighlight).props.onPress();

    expect(spy).toBeCalled();
  });
});
