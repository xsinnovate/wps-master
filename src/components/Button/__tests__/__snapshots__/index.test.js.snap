// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "borderColor": "#333",
        "borderRadius": 20,
        "borderWidth": 1,
        "height": 40,
        "paddingHorizontal": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 40,
          "textAlign": "center",
        }
      }
    >
      Label
    </Text>
  </View>
</View>
`;
